content/articles/data-preprocessing-python/index.md

```diff
@@ -0,0 +1,248 @@
+
+The `Country` and the Purchased columns of our dataset contain data that fall into categories. Since machine learning models are based on the mathematical equation, which takes only numerical inputs, it will be difficult for the machine learning model to compute the correlation between the feature and the dependent variables. To ensure this does not happen, we need to convert those string entries into numbers, i.e., we shall encode France into 0, Spain into 1, and Germany into 2. However, our future machine learning model can interpret the numerical order between 0 for France, 1 for Spain, and 2 for Germany do matter, which is not the case. To ensure this misinterpretation does not occur, we make use of one-hot encoding. One-hot encoding converts ours country column into three columns and creates a binary vector for each country such that there is no numerical order between the country categories. Therefore, instead of encoding the country variable into 1,2, and 3, vectors [1, 0 ,0] for the France, [0, 1, 0] for the Spain and [0, 0, 1] for the Germany. Similarly, we shall encode the Purchased column into 1,s for Yes and 0,s for No.
+
+First, let us start with encoding the country column, which falls on the features subset x.
+
+To achieve this, let us execute the code below.
+
+```python
+from sklearn.compose import ColumnTransformer
+from sklearn.preprocessing import OneHotEncoder
+ct = ColumnTransformer(transformers=[('encoder', OneHotEncoder(), [0])], remainder= 'passthrough')
+x = np.array(ct.fit_transform(x))
+```
+**Output**
+```Python
+# executing the cell we obtain:
+print(x)
+```
+![Encoded Country column](/engineering-education/data-preprocessing-python/Encoding_country.png)
+
+  From the output,the country column has been transformed into 3 columns with each row representing only one encoded column where, France was encoded into a vector [1.0 0.0 0.0], Spain encoded into a vector [0.0 0.0 1.0], and Germany encoded into a vector [0.0 1.0 0.0] and they're all unique.
+
+To encode our depended variable y, let's run the code below.
+
+ ```python
+ from sklearn.preprocessing import LabelEncoder
+le = LabelEncoder()
+y = le.fit_transform(y)
+```
+
+**Output**
+```Python
+print(y)
+```
+```python
+[0 1 0 0 1 1 0 1 0 1]
+
+```
+Our dependent variable is encoded successfully into 0's and 1's.
+
+### Step 5: Splitting the dataset into the training and test set
+
+In machine learning, we split the dataset into a training set and test set. The training set is the fraction of a dataset that we use to implement the model. On the other hand,  the test set is the fraction of the dataset that we use to evaluate the performance our the model. The test set is assumed to be unknown during the process of the model implementation.
+We need to split our dataset into four subsets, i.e., x_train, x_test, y_train, and y_test. 
```

Comment:
`x_train`, `x_test`… Use these throughout the article, whenever a variable is referenced.